react/components/SKUSelector/components/SelectVariationMode.tsx

```diff
@@ -0,0 +1,55 @@
+import styles from '../styles.css'
+import { DisplayOption } from '../types'
+import React, { useCallback } from 'react'
+import { Dropdown } from 'vtex.styleguide'
+import { injectIntl, InjectedIntlProps, defineMessages } from 'react-intl'
+
+interface VariationSelectModeProps extends InjectedIntlProps {
+  selectedItem: string | null
+  displayOptions: DisplayOption[]
+}
+
+const messages = defineMessages({
+  selectPlaceholder: {
+    id: 'store/sku-selector.select.placeholder',
+    defaultMessage: '',
+  }
+})
+
+function SelectVariationMode(props: VariationSelectModeProps) {
+  const {
+    intl,
+    selectedItem,
+    displayOptions,
+  } = props
+
+  const options = displayOptions.map(op => ({
+    label: op.label,
+    value: op.label,
+  }))
+
+  const mapOptions = useCallback(() => {
```

Comment:
I think you could just inline this code inside the `handleClick`, the usage of `useCallback` here won't be that useful.

Also, if you're creating a new object using reduce prefer doing mutations, instead of creating new objects each iteration, since those intermediate objects will be just garbage collected later on.

```diff
- (acc, cur) => ({ ...acc, [cur.label]: cur })
+ (acc, cur) => {
+  acc[cur.label] = cur;
+  return acc;
+ }
```